core/domain/summary_services.py

```diff
@@ -163,6 +164,74 @@ def get_displayable_collection_summary_dicts_matching_ids(collection_ids):
     return _get_displayable_collection_summary_dicts(collection_summaries)
 
 
+def get_exp_metadata_dicts_matching_query(query_string, search_cursor, user_id):
+    """Given a query string and a search cursor, returns a list of exploration
+    metadata dicts that satisfy the search query.
+
+    Args:
+        query_string: str. The search query for which the search is to be
+            performed.
+        search_cursor: str or None. The cursor location to start the search
+            from. If None, the returned values are from the beginning
+            of the results list.
+        user_id: str. The user id affect the result by restricting the results
```

Comment:
This is still a bit unclear. Follow the docstring that you used for get_exploration_metadata_dicts, which says exactly how the user_id affects the result.